<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Provides a convenient way to analyze and visualize PICRUSt2 output with pre-defined plots and functions. Allows for generating statistical plots about microbiome functional predictions and offers customization options. Features a one-click option for creating publication-level plots, saving time and effort in producing professional-grade figures. Streamlines the PICRUSt2 analysis and visualization process. For more details, see Yang et al. (2023) &lt;arXiv:2303.10388&gt;.">
<title>Make PICRUSt2 Output Analysis and Visualization Easier • ggpicrust2</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Make PICRUSt2 Output Analysis and Visualization Easier">
<meta property="og:description" content="Provides a convenient way to analyze and visualize PICRUSt2 output with pre-defined plots and functions. Allows for generating statistical plots about microbiome functional predictions and offers customization options. Features a one-click option for creating publication-level plots, saving time and effort in producing professional-grade figures. Streamlines the PICRUSt2 analysis and visualization process. For more details, see Yang et al. (2023) &lt;arXiv:2303.10388&gt;.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">ggpicrust2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.8</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/cafferychen777/ggpicrust2/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="ggpicrust2">ggpicrust2<a class="anchor" aria-label="anchor" href="#ggpicrust2"></a>
</h1></div>
<p>ggpicrust2 is a comprehensive package that integrates pathway name/description annotations, ten of the most advanced differential abundance (DA) methods, and visualization of DA results. It offers a comprehensive solution for analyzing and interpreting the results of picrust2 functional prediction in a seamless and intuitive way. Whether you are a researcher, data scientist, or bioinformatician, ggpicrust2 can help you better understand the underlying biological processes and mechanisms at play in your picrust2 output data. So if you are interested in exploring the output data of picrust2, ggpicrust2 is the tool you need.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of ggpicrust2 from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"cafferychen777/ggpicrust2"</span><span class="op">)</span></span></code></pre></div>
<p>We are actively preparing to upload the package to bioconductor.</p>
</div>
<div class="section level2">
<h2 id="workflow">Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h2>
<p>The easiest way to analyze the picrust2 output is using ggpicrust2() function. The entire pipeline can be run with ggpicrust2() function.</p>
<p>ggpicrust2() integrates ko abundance to kegg pathway abundance conversion, annotation of pathway, differential abundance (DA) analysis, DA results visualization.</p>
<p><img src="https://cafferychen777.github.io/ggpicrust2/reference/figures/ggpicrust2%20fig1.jpeg"></p>
<p><img src="https://cafferychen777.github.io/ggpicrust2/reference/figures/ggpicrust2%20fig2.jpeg"></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#If you want to analysis kegg pathway abundance instead of ko within the pathway. You should turn ko_to_kegg to TRUE.</span></span>
<span><span class="co">#The kegg pathway typically have the more explainable description.</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">read_delim</span><span class="op">(</span></span>
<span>    <span class="st">"~/Microbiome/C9orf72/Code And Data/new_metadata.txt"</span>,</span>
<span>    delim <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>    escape_double <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    trim_ws <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">group</span> <span class="op">&lt;-</span> <span class="st">"Enviroment"</span></span>
<span><span class="va">daa_results_df</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="reference/ggpicrust2.html">ggpicrust2</a></span><span class="op">(</span></span>
<span>    file <span class="op">=</span> <span class="st">"/Users/apple/Microbiome/C9orf72/Code And Data/picrust2_out/KO_metagenome_out/pred_metagenome_unstrat.tsv/pred_metagenome_unstrat.tsv"</span>,</span>
<span>    metadata <span class="op">=</span> <span class="va">metadata</span>,</span>
<span>    group <span class="op">=</span> <span class="st">"Enviroment"</span>,</span>
<span>    pathway <span class="op">=</span> <span class="st">"KO"</span>,</span>
<span>    daa_method <span class="op">=</span> <span class="st">"ALDEx2"</span>,</span>
<span>    order <span class="op">=</span> <span class="st">"pathway_class"</span>,</span>
<span>    ko_to_kegg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    x_lab <span class="op">=</span> <span class="st">"pathway_name"</span>,</span>
<span>    p.adjust <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>    select <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    reference <span class="op">=</span> <span class="cn">NULL</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#The visualization will be published in viewer.</span></span>
<span></span>
<span><span class="co">#If you want to analysis the EC. MetaCyc. KO without conversions. You should turn ko_to_kegg to FALSE.</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">read_delim</span><span class="op">(</span></span>
<span>    <span class="st">"~/Microbiome/C9orf72/Code And Data/new_metadata.txt"</span>,</span>
<span>    delim <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>    escape_double <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    trim_ws <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">group</span> <span class="op">&lt;-</span> <span class="st">"Enviroment"</span></span>
<span><span class="va">daa_results_df</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="reference/ggpicrust2.html">ggpicrust2</a></span><span class="op">(</span></span>
<span>    file <span class="op">=</span> <span class="st">"//Users/apple/Microbiome/C9orf72/Code And Data/picrust2_out/EC_metagenome_out/pred_metagenome_unstrat.tsv/pred_metagenome_unstrat.tsv"</span>,</span>
<span>    metadata <span class="op">=</span> <span class="va">metadata</span>,</span>
<span>    group <span class="op">=</span> <span class="st">"Enviroment"</span>,</span>
<span>    pathway <span class="op">=</span> <span class="st">"EC"</span>,</span>
<span>    daa_method <span class="op">=</span> <span class="st">"ALDEx2"</span>,</span>
<span>    order <span class="op">=</span> <span class="st">"pathway_class"</span>,</span>
<span>    ko_to_kegg <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    x_lab <span class="op">=</span> <span class="st">"description"</span>,</span>
<span>    p.adjust <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>    select <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    reference <span class="op">=</span> <span class="cn">NULL</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#The visualization will be published in viewer.</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="output">Output<a class="anchor" aria-label="anchor" href="#output"></a>
</h2>
<p>The typical output of the ggpicrust2 is like this.</p>
<p><img src="https://cafferychen777.github.io/ggpicrust2/reference/figures/pathway_errorbar.jpg"></p>
<p>You’ll still need to render <code>README.Rmd</code> regularly, to keep <code>README.md</code> up-to-date. <code><a href="https://devtools.r-lib.org/reference/build_rmd.html" class="external-link">devtools::build_readme()</a></code> is handy for this. You could also use GitHub Actions to re-render <code>README.Rmd</code> every time you push. An example workflow can be found here: <a href="https://github.com/r-lib/actions/tree/v1/examples" class="external-link uri">https://github.com/r-lib/actions/tree/v1/examples</a>.</p>
</div>
<div class="section level2">
<h2 id="function-details">function details<a class="anchor" aria-label="anchor" href="#function-details"></a>
</h2>
<div class="section level3">
<h3 id="ko2kegg_abundance">ko2kegg_abundance()<a class="anchor" aria-label="anchor" href="#ko2kegg_abundance"></a>
</h3>
<p>KEGG Orthology(KO) is a classification system developed by the Kyoto Encyclopedia of Genes and Genomes (KEGG) data-base(Kanehisa et al., 2022). It uses a hierarchical structure to classify enzymes based on the reactions they catalyze. To better understand pathways’ role in different groups and classify the pathways, the KO abundance table needs to be converted to KEGG pathway abundance. But PICRUSt2 removes the function from PICRUSt. ko2kegg_abundance() can help convert the table.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Sample usage of the ko2kegg_abundance function</span></span>
<span></span>
<span><span class="co"># Assume that the KO abundance table is stored in a file named "ko_abundance.tsv"</span></span>
<span></span>
<span><span class="va">ko_abundance_file</span> <span class="op">&lt;-</span> <span class="st">"ko_abundance.tsv"</span></span>
<span></span>
<span><span class="va">kegg_abundance</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ko2kegg_abundance.html">ko2kegg_abundance</a></span><span class="op">(</span><span class="va">ko_abundance_file</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The resulting kegg_abundance data frame can now be used for further analysis and visualization.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pathway_daa">pathway_daa()<a class="anchor" aria-label="anchor" href="#pathway_daa"></a>
</h3>
<p>Differential abundance(DA) analysis plays a major role in PICRUSt2 downstream analysis. pathway_daa() integrates almost all DA methods applicable to the predicted functional profile which there excludes ANCOM and ANCOMBC. It includes ALDEx2(Fernandes et al., 2013), DEseq2(Love et al., 2014), Maaslin2(Mallick et al., 2021), Lin-DA(Zhou et al., 2022), edgeR(Robinson et al., 2010) , limma voom(Ritchie et al., 2015), metagenomeSeq(Paulson et al., 2013), lefser(Segata et al., 2011).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#the abundance table is better to be data.frame rather than tibble</span></span>
<span><span class="co">#you can use ko2_kegg_abundance output</span></span>
<span><span class="va">abundance</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ko2kegg_abundance.html">ko2kegg_abundance</a></span><span class="op">(</span><span class="va">ko_abundance_file</span><span class="op">)</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">read_delim</span><span class="op">(</span></span>
<span>    <span class="st">"~/Microbiome/C9orf72/Code And Data/new_metadata.txt"</span>,</span>
<span>    delim <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>    escape_double <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    trim_ws <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">group</span> <span class="op">&lt;-</span> <span class="st">"Enviroment"</span></span>
<span><span class="co">#daa_method default is "ALDex2"</span></span>
<span><span class="va">daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">abundance</span>,</span>
<span>           metadata <span class="op">=</span> <span class="va">metadata</span>,</span>
<span>           group <span class="op">=</span> <span class="va">group</span> ,</span>
<span>           daa_method <span class="op">=</span> <span class="st">"limma voom"</span>,</span>
<span>           select <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>           p.adjust <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>           reference <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#If you group levels &gt;3 and want to use the LinDA or limma voom, you should give a reference.</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">read_delim</span><span class="op">(</span></span>
<span>    <span class="st">"~/Microbiome/C9orf72/Code And Data/new_metadata.txt"</span>,</span>
<span>    delim <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>    escape_double <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    trim_ws <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">group</span> <span class="op">&lt;-</span> <span class="st">"Group"</span></span>
<span><span class="va">daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_daa.html">pathway_daa</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">abundance</span>,</span>
<span>           metadata <span class="op">=</span> <span class="va">metadata</span>,</span>
<span>           group <span class="op">=</span> <span class="va">group</span> ,</span>
<span>           daa_method <span class="op">=</span> <span class="st">"limma voom"</span>,</span>
<span>           select <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>           p.adjust <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>           reference <span class="op">=</span> <span class="st">"Harvard BRI"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pathway_annotation">pathway_annotation<a class="anchor" aria-label="anchor" href="#pathway_annotation"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">daa_results_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_annotation.html">pathway_annotation</a></span><span class="op">(</span>pathway <span class="op">=</span> <span class="st">"KO"</span>, daa_results_df <span class="op">=</span> <span class="va">daa_results_df</span>, ko_to_kegg <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pathway_errorbar">pathway_errorbar<a class="anchor" aria-label="anchor" href="#pathway_errorbar"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/pathway_errorbar.html">pathway_errorbar</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">abundance</span>,</span>
<span>           daa_results_df <span class="op">=</span> <span class="va">daa_results_df</span>,</span>
<span>           Group <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">Enviroment</span>,</span>
<span>           ko_to_kegg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>           p_values_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>           order <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>           select <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>           p_value_bar <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>           colors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>           x_lab <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pathway_heatmap">pathway_heatmap<a class="anchor" aria-label="anchor" href="#pathway_heatmap"></a>
</h3>
<p>pathway_heatmap() can visualize the patterns in PICRUSt2 output data, which can be useful for identifying trends or highlighting areas of in-terest.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abundance</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ko2kegg_abundance.html">ko2kegg_abundance</a></span><span class="op">(</span><span class="va">ko_abundance_file</span><span class="op">)</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">read_delim</span><span class="op">(</span></span>
<span>    <span class="st">"~/Microbiome/C9orf72/Code And Data/new_metadata.txt"</span>,</span>
<span>    delim <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>    escape_double <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    trim_ws <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">group</span> <span class="op">&lt;-</span> <span class="st">"Enviroment"</span></span>
<span><span class="fu"><a href="reference/pathway_heatmap.html">pathway_heatmap</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">abundance</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="va">group</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pathway_pca">pathway_pca()<a class="anchor" aria-label="anchor" href="#pathway_pca"></a>
</h3>
<p>pathway_pca() can show the difference after dimensional reduction by PCA.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># generate example abundance matrix and metadata dataframe</span></span>
<span><span class="va">abundance</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">200</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>                       Treatment <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"X"</span>, <span class="st">"Y"</span><span class="op">)</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># perform PCA and create plot</span></span>
<span><span class="va">pathway_pca_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pathway_pca.html">pathway_pca</a></span><span class="op">(</span>abundance <span class="op">=</span> <span class="va">abundance</span>, metadata <span class="op">=</span> <span class="va">metadata</span>, group <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">pathway_pca_plot</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=ggpicrust2" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/cafferychen777/ggpicrust2/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/cafferychen777/ggpicrust2/pulls" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing ggpicrust2</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Chen Yang <br><small class="roles"> Author, maintainer </small>  </li>
<li>Liangliang Zhang <br><small class="roles"> Author </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Chen Yang, Liangliang Zhang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
